cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -130,30 +139,16 @@ public void destroy() {
     // Nothing to do
   }
 
-  private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         CConfiguration cConf) {
-    Preconditions.checkState(port > 0, "Port number is invalid.");
-    int numPartitions = cConf.getInt(KafkaConstants.ConfigKeys.NUM_PARTITIONS_CONFIG_DEPRECATED);
-    Preconditions.checkState(numPartitions > 0, "Num partitions should be greater than zero.");
-
+  private Properties generateKafkaConfig(CConfiguration cConf) {
     Properties prop = new Properties();
 
-    prop.setProperty("broker.id", Integer.toString(brokerId));
-    if (hostname != null) {
-      prop.setProperty("host.name", hostname);
-    }
-    prop.setProperty("port", Integer.toString(port));
-    prop.setProperty("zookeeper.connect", zkConnectStr);
-
-    //ultimately want regex to be "^(kafka\\..server\\.*)$" but for now this gets all deprecated and current config keys
-    //since all deprecated key accesses are redirected to the current config keys in deprecatedKeyMap in Configuration
-    Map<String, String> propConfigs = cConf.getValByRegex("^(kafka\\..*)$");
+    Map<String, String> propConfigs = cConf.getValByRegex("^(kafka\\..server\\.*)$");
```

Comment:
So now, if a user uses the deprecated key, we will be able to look it up via the new key here?
Example:
old key = kafka.param
new key = kafka.server.param
user defines kafka.param to be 10, in their cdap-site.xml
Will cConf.get('kafka.server.param') return 10?

If not, how does this handle deprecated keys?